lib/client/db/dbcmd/error.go

```diff
@@ -46,6 +46,12 @@ func ConvertCommandError(cmd *exec.Cmd, err error, peakStderr string) error {
 		}
 	}
 
+	lowerCaseStderr := strings.ToLower(peakStderr)
+	if strings.Contains(lowerCaseStderr, "access to db denied") {
+		fmtString := "'%s' exited with the error above. Use 'tsh db ls' to see your available logins, " +
```

Comment:
I think that the error message is a bit confusing. I understand the first part, but why should a user ask the Teleport admin to check the audit log? I'd say if you're missing access, you should ask the admin to add the missing role/resource to your account. You could also add a link to our docs with the access request, but I'm not sure if this is something we want to do, link: https://goteleport.com/docs/access-controls/access-requests/.
cc @smallinsky

Reply:
Thinking about this more, I think I was getting ahead of myself with being "helpful" by assuming the Teleport admin would not know why the user was denied access 😄. How's this phrasing: "... or ask your Teleport administrator to grant you access."

Reply:
The error message looks fine thought can we create a db access troubleshooting entry and reference link to it similar like in https://github.com/gravitational/teleport/pull/14794/files#diff-f9a594b642a0869ab30233d33329c066571f4ecba227e8e32b30ca606beb0ccfR10 ? 

The access-denied can also occur when:
- Teleport RBAC allows to connect with a user x  but user x is not  created in db instance. 
- RBAC model allows to connect with a user x but db_name RBAC validation failed. 

This `access to db denied` tsh db connect error mostly confuses a new teleport users where they don't know to much about teleport RBAC model.

Reply:
Shall I simply link to https://goteleport.com/docs/database-access/troubleshooting (which will exist once #14794 merges) and open a PR against #14794 to add a section to the new troubleshooting section for RBAC?

Reply:
```diff
  > tsh db connect --db-user=baduser teleport-mysql
  Tap any security key
  ERROR 1105 (HY000): access to db denied
  ERROR: exit status 1: '/opt/homebrew/bin/mysql' exited with the above error. Use 'tsh db ls' to see your available logins, or ask your Teleport administrator to grant you access. 
+ See https://goteleport.com/docs/database-access/troubleshooting for more information.
```

Reply:
> Shall I simply link to https://goteleport.com/docs/database-access/troubleshooting (which will exist once https://github.com/gravitational/teleport/pull/14794 merges) and open a PR against https://github.com/gravitational/teleport/pull/14794 to add a section to the new troubleshooting section for RBAC?

We should link to specific troubleshooting header like:  #certificate-expired-or-is-not-yet-valid but to do that the https://github.com/gravitational/teleport/pull/14794 needs to be merged.

Reply:
Updated to reference a new heading that will be added in #17259

Reply:
#17259 is approved and should auto-merge soon. @smallinsky I think this is ready if you want to approve

Reply:
#17259 merged